model/credentials/db.go

```diff
@@ -0,0 +1,200 @@
+package credentials
+
+import (
+	"context"
+	"time"
+
+	"github.com/evergreen-ci/certdepot"
+	"github.com/evergreen-ci/evergreen"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/jasper/rpc"
+	"github.com/pkg/errors"
+	"github.com/square/certstrap/depot"
+)
+
+const (
+	Collection  = "credentials"
+	CAName      = "evergreen"
+	ServiceName = "evergreen.mongodb.com"
+
+	defaultConnectionTimeout = 10 * time.Second
+	defaultExpiration        = 30 * 24 * time.Hour // 1 month
+)
+
+// Bootstrap performs one-time initialization of the credentials collection with
+// the certificate authority and evergreen service certificate.
+func Bootstrap() error {
+	env := evergreen.GetEnvironment()
+	settings := env.Settings()
+	maxExpiration := time.Duration(1<<63 - 1)
+
+	caConfig := certdepot.CertificateOptions{
+		CA:         CAName,
+		CommonName: CAName,
+		Expires:    maxExpiration,
+	}
+
+	serviceConfig := certdepot.CertificateOptions{
+		CA:         CAName,
+		CommonName: ServiceName,
+		Host:       ServiceName,
+		Expires:    maxExpiration,
+	}
+
+	bootstrapConfig := certdepot.BootstrapDepotConfig{
+		MongoDepot:  mongoConfig(settings),
+		CAName:      CAName,
+		CAOpts:      &caConfig,
+		ServiceName: ServiceName,
+		ServiceOpts: &serviceConfig,
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), defaultConnectionTimeout)
+	defer cancel()
+
+	if _, err := certdepot.BootstrapDepotWithMongoClient(ctx, env.Client(), bootstrapConfig); err != nil {
+		return errors.Wrap(err, "could not bootstrap credentials collection")
+	}
+
+	return nil
+}
+
+func mongoConfig(settings *evergreen.Settings) *certdepot.MongoDBOptions {
+	return &certdepot.MongoDBOptions{
+		MongoDBURI:     settings.Database.Url,
+		DatabaseName:   settings.Database.DB,
+		CollectionName: Collection,
+	}
+}
+
+// getDepot returns the certificate depot connected to the database.
+func getDepot(ctx context.Context, env evergreen.Environment) (depot.Depot, error) {
+	return certdepot.NewMongoDBCertDepotWithClient(ctx, env.Client(), mongoConfig(env.Settings()))
+}
+
+func deleteIfExists(dpt depot.Depot, tag *depot.Tag) error {
+	if dpt.Check(tag) {
+		return errors.Wrap(dpt.Delete(tag), "problem deleting existing tag")
+	}
+	return nil
+}
+
+// SaveCredentials saves the credentials from the options for the given user
+// name. If the credentials already exist, this will overwrite the existing
+// credentials.
+func SaveCredentials(name string, creds *rpc.Credentials) error {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultConnectionTimeout)
+	defer cancel()
+
+	dpt, err := getDepot(ctx, evergreen.GetEnvironment())
```

Comment:
I would avoid summoning the environment here, environment should generally be passed in.